### PR TITLE
feat: typed store/orders

### PR DESCRIPTION
**Pull request by kasperkristensen**

### CODE REVIEW

File: packages/medusa-js/src/index.ts [hunk 1]
```diff
@@ -1,45 +1,45 @@
...
+import MedusaError from './error';
+import Client, { Config } from './request';
+import AuthResource from './resources/auth';
+import CartsResource from './resources/carts';
+import CustomersResource from './resources/customers';
+import OrdersResource from './resources/orders';
+import ProductsResource from './resources/products';
+import RegionsResource from './resources/regions';
+import ReturnReasonsResource from './resources/return-reasons';
+import ReturnsResource from './resources/returns';
```
Comment (olivermrbl) [inline packages/medusa-js/src/index.ts:10]: Needs linting fix

File: packages/medusa-js/src/resources/orders.ts [hunk 2]
```diff
@@ -1,33 +1,34 @@
...
+import { AxiosPromise } from 'axios';
 import BaseResource from './base';
-import * as Types from '../types';
 
 class OrdersResource extends BaseResource {
   /**
    * @description Retrieves an order
-   * @param id is required
-   * @returns AsyncResult<{ order: Order }>
+   * @param {string} id is required
```
Comment (olivermrbl) [inline packages/medusa-js/src/resources/orders.ts:8]: General comment, the client should use a `StoreOrdersRes` as response type :)

File: packages/medusa/src/api/routes/store/orders/index.ts [hunk 3]
```diff
@@ -105,3 +107,9 @@ export const allowedFields = [
   "subtotal",
   "total",
 ]
+
+export type StoreOrderResponse = {
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/store/orders/index.ts:111]:
```suggestion
export type StoreOrderRes = {
```

File: packages/medusa/src/api/routes/store/orders/lookup-order.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  res.json({ order })
+}
+
+export class ShippingAddressPayload {
+  @IsOptional()
+  @IsString()
+  postal_code: string
+}
+
+export class StoreGetOrdersReq {
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/store/orders/lookup-order.ts:65]:
```suggestion
export class StoreGetOrdersParams {
```

**Review by olivermrbl — CHANGES_REQUESTED** [review]

**Comment by kasperkristensen** [comment]
Moved to [#803](https://github.com/medusajs/medusa/pull/803)

**Pull request closed without merging**